gitimerge.py

```diff
@@ -4137,11 +4136,15 @@ def add_tip2_argument(subparser):
         parser.error('Unrecognized subcommand')
 
 
-if __name__ == '__main__':
+def climain():
     try:
-        main(sys.argv[1:])
+        main()
```

Comment:
It took me a while to figure out how the command-line arguments are getting through to `main`. (I wondered about that [here](https://github.com/mhagger/git-imerge/pull/143#discussion_r290060057).) It used to be that they were passed to `main` here. But now, nothing is passed to `main`, and nothing anywhere in the script references `sys.argv` anymore.

It turns out that we are relying on [this call to `parser.parse_args()`](https://github.com/mhagger/git-imerge/blob/224ce4c17d79730046b0cc2c8a61c28cf9366403/gitimerge.py#L4096), which now passes `None` to that method, which triggers `argparse`'s default behavior of taking the arguments directly from `sys.argv`.

This seems a bit magical to me. Is there a reason not to continue passing `sys.argv[1:]` explicitly here, as was done before? ("This is the standard pattern now" would an acceptable answer.) Otherwise, could `main` get a short docstring explaining that `args = None` implies that the arguments will be taken from `sys.argv`?

Otherwise, everything looks great! Thanks for your patience :smiley: